ex04.ex

```diff
@@ -74,11 +82,16 @@ defmodule Ex04 do
   helps you do that. And, if you use that function, what does it return? That
   return value will be the thing you have to manipulate.
   """
-
-  def even_odd . . . "your code"
-
-
-
+  def even_odd([ h | t ]) do
```

Comment:
## -1

same comment as above.

Also (no deduction for this) try to avoid ++. It turns out to be faster to use `[ a | tail ]` to build the list, and then reverse either the original input or the final result(s).

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~